pkg/codegen/docs/constructor_syntax_generator.go

```diff
@@ -86,6 +86,9 @@ func (g *constructorSyntaxGenerator) writeValue(
 		write("fileArchive(\"./path/to/archive\")")
 	case schema.AssetType:
 		write("stringAsset(\"content\")")
+	case schema.AnyResourceType:
+		// Constructors are language specific - just "resource?"
+		write("resource")
```

Comment:
constructor syntax logic is currently _skipping_ over properties that are typed as resources, see `isResourceType` function. Because this will just fail when binding the PCL program saying that the identifier `resource` is not defined anywhere

Reply:
This `pulumi.json#/Resource` is more like an asset or archive, an opaque type, so docs gen does see this down below:

https://github.com/pulumi/pulumi/blob/fbf8cacabdb6eb620a64eb384481da3d3666eaaa/tests/testdata/codegen/any-resource/docs/resourcewithresources/_index.md

Reply:
Aahh I guess this works because we use non-strict binding for the constructor syntax 😅 my bad! This LGTM ✅